src/stash/src/tests.rs

```diff
@@ -570,9 +570,11 @@ async fn test_stash_batch_large_number_updates() {
     Stash::with_debug_stash(|mut stash| async move {
         let col = collection::<i64, i64>(&mut stash, "c1").await.unwrap();
         let mut batch = make_batch(&col, &mut stash).await.unwrap();
-        for i in 0..500_000 {
-            col.append_to_batch(&mut batch, &i, &(i + 1), 1);
-        }
+        let batches: Vec<_> = (0..500_00).map(|i| (i, (i + 1), 1)).collect();
```

Comment:
You are making the test do 10x less stuff here, no? I'm suspicious that could be why you saw a 10x gain ...

Reply:
> You are making the test do 10x less stuff here, no?

I believe so. The previous code had 500,000 calls to `append` on a `Vec`, which would result in `log_2(500000) = 18` allocations. The new code only has 2 allocations, the call to `collect()` and the call to `extend()`, so it is almost 10x less work. 

I'll try running the test a couple more times. Maybe the runtime is just very inconsistent.

Reply:
My point is that you changed `500_000` (i.e., 500 thousand) to `500_00` (i.e., 50 thousand), which looks like a mistake, unless I'm misunderstanding something.

Reply:
Oooh, I completely missed that, thanks for pointing it out. I'll update the test with the extra 0.

Reply:
Yeah adding the extra 0 back removes all speed gains locally and I'd guess in persist too.

Reply:
Yep, it now takes 38 seconds instead of 44 seconds which doesn't seem like a big difference.